src/Groups/GAPGroups.jl

```diff
@@ -1448,6 +1499,29 @@ false
 @gapattribute is_finitelygenerated(G::GAPGroup) = GAP.Globals.IsFinitelyGeneratedGroup(G.X)::Bool
 
 
+@doc raw"""
+    is_free(G::GAPGroup)
+
+Return whether `G` is a free group.
```

Comment:
We should say what is meant with "is a free group".
GAP's `IsFreeGroup` means essentially a group created with `free_group` or a subgroup of such a group.
The GAP documentation is also vague about this, but it states that groups consisting of elements in the filter `IsAssocWordWithInverse` are regarded as free groups.

One could have the idea that "free group" means a group that is free on some set of elements, and then each trivial group is free on the empty set, and certain factors of free groups (for example where just some generators are mapped to the identity) are also free.

Reply:
Yes, I think that the "technical interpretation" is used more frequently; also GAP's `IsFreeGroup` is just a filter, there was no aim to support a check for the mathematical property.

So do we need a better name for the "technical interpretation", in order to keep `is_free` for the mathematical one,
or do we decide that we introduce the type `FreeGroup` as mentioned above?

(I guess the name `IsFreeGroup` was chosen because people did not think that checking the mathematical property for arbitrary f. p. groups would be interesting. Moreover, as soon as such a function would be available, somebody would come and ask for a function that checks whether a group is free on a given set, and for a function that returns a set on which a given group is free ...)